common/djangoapps/student/models.py

```diff
@@ -2145,7 +2147,11 @@ def _get_enrollment_state(cls, user, course_key):
         if not enrollment_state:
             try:
                 record = cls.objects.get(user=user, course_id=course_key)
-                enrollment_state = CourseEnrollmentState(record.mode, record.is_active)
+                # Filter per organization.
+                if not next(filter_enrollments([record]), False):
+                    enrollment_state = CourseEnrollmentState(None, None)
+                else:
+                    enrollment_state = CourseEnrollmentState(record.mode, record.is_active)
             except cls.DoesNotExist:
```

Comment:
```suggestion
                # Filter per organization.
                record = next(filter_enrollments([record]))
                enrollment_state = CourseEnrollmentState(record.mode, record.is_active)
            except cls.DoesNotExist, StopIteration:
```

Can you test if this works? I think this is more legible, but if you think otherwise let me know.

Reply:
@MoisesGSalas It works :) Thanks for the suggestion.